src/number/n_unknown.jl

```diff
@@ -242,8 +242,116 @@ end
 #
 ###############################################################################
 
+mutable struct MutableRingWrapper{S, T} <: Nemo.Ring
+   data::S
+end
+
+mutable struct MutableRingElemWrapper{S, T} <: Nemo.RingElem
+   data::T
+   parent::MutableRingWrapper{S, T}
+end
+
+function promote_rule(::Type{n_unknown{MutableRingElemWrapper{S, T}}}, ::Type{T}) where {T <: Nemo.RingElem, S}
+   return T
+end
+
+function expressify(a::MutableRingWrapper{S, T}; context = nothing) where {S, T}
+   return expressify(a.data, context = context)
+end
+
+function expressify(a::MutableRingElemWrapper{S, T}; context = nothing) where {S, T}
+   return expressify(a.data, context = context)
+end
+
+function Base.show(io::IO, a::MutableRingWrapper{S, T}) where {S, T}
+   Base.show(io, a.data)
+end
+
+function Base.show(io::IO, a::MutableRingElemWrapper{S, T}) where {S, T}
+   Base.show(io, a.data)
+end
+
+function elem_type(::Type{MutableRingWrapper{S, T}}) where {S, T}
+   return MutableRingElemWrapper{S, T}
+end
+
+function parent(a::MutableRingElemWrapper{S, T}) where {S, T}
+   return a.parent
+end
+
+function (R::MutableRingWrapper{S, T})() where {S, T}
+   return MutableRingElemWrapper{S, T}(R.data(), R)
+end
+
+function (R::MutableRingWrapper{S, T})(a::MutableRingElemWrapper) where {S, T}
+   @assert R == a.parent
+   return MutableRingElemWrapper{S, T}(R.data(a.data), R)
+end
+
+function Base.deepcopy_internal(a::MutableRingElemWrapper{S, T}, dict::IdDict) where {S, T}
+   return MutableRingElemWrapper{S, T}(deepcopy_internal(a.data, dict), a.parent)
+end
+
+# should be R::S
+function (R::Nemo.Ring)(a::n_unknown{MutableRingElemWrapper{S, T}}) where {S, T}
```

Comment:
this is too general and R::S doesn't work.

Reply:
This cannot be implement in julia. It is a limitation of the current implementation.

The problem is that adding this here makes `(::PolyRing{n_unkown{Mutable..}})(x::n_unkown{Mutable...})` ambiguous, since the new method is more specific in the second argument and the old `(::PolyRing{T})(::T)` is more specific in the first argument. Can you add the `(::PolyRing{n_unkown{Mutable...}})(x::n_unkown{Mutable...})` method and see if it works?

Reply:
works, but I don't like it very much